### PR TITLE
chore(helper): let user specify version in docker helper

### PR DESCRIPTION
**Pull request by mrjones-plip**
# Description

This ticket allows the docker helper to either default to most recent CHT version (current behavior) or allow the user to select from all known versions.

This PR should accompany a TBD docs PR so users will know about the feature

## testing

1. check out this branch `docker-helper-specify-version`
2. run through happy path of creating a new project - ensure it's on `4.3.0` (latest as of now
3. ensure existing project will start with BOTH `./cht-docker-compose.sh ENV_FILE_HERE` and following prompts to specify an existing project
4. run through new path and ensure new project of older version (branch of 4.x.x) works

# Code review checklist

- [X] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [X] Tested: Unit and/or e2e where appropriate
- [X] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# Compose URLs

If Build CI hasn't passed, these may 404:

* [Core](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:docker-helper-specify-version/docker-compose/cht-core.yml)
* [CouchDB Single](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:docker-helper-specify-version/docker-compose/cht-couchdb.yml)
* [CouchDB Cluster](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:docker-helper-specify-version/docker-compose/cht-couchdb-clustered.yml)

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by mrjones-plip** [comment]
@njogz  - feel like reviewing this PR now that you're docker helper power user :stuck_out_tongue_closed_eyes:  ?

**Comment by tatilepizs** [comment]
Hi @mrjones-plip, 
Thank you for this improvement to the docker-helper, it is great.

I was testing it, just out of curiosity 🙃 and I wanted to ask you. What do you think about asking about `cht-core` version when you start an instance that was already created? do you think that it might be confusing? 

I am asking because sometimes I use the same instance to test a ticket, so first I need to test it in `master` and then in the branch that it was fixed, so I was thinking that maybe it could be a good idea if, once I tested it in `master`, for example, I stop the instance, and once I start it again it can ask me if I want to continue using the same `cht-core` version, in this case `master`, or if I want to change it.

What do you think?

File: scripts/docker-helper-4.x/cht-docker-compose.sh [hunk 1]
```diff
@@ -46,14 +47,27 @@ CHT_NETWORK=$projectName-cht-net
 EOL
 }
 
-get_latest_release() {
-  latest=$(curl -s https://staging.dev.medicmobile.org/_couch/builds_4/_design/builds/_view/releases\?limit\=1\&descending\=true |  tr -d \\n | grep -o 'medic\:medic\:[0-9\.]*')
-  echo "https://staging.dev.medicmobile.org/_couch/builds_4/${latest}"
+get_compose_download_url() {
+	preferredRelease=$1
+	if [ -z "$preferredRelease" ]; then
```
Comment (garethbowen) [inline scripts/docker-helper-4.x/cht-docker-compose.sh:52]: I think you've got a mix of tabs and spaces which is leading to indentation issues here.

File: scripts/docker-helper-4.x/cht-docker-compose.sh [hunk 2]
```diff
@@ -261,14 +289,15 @@ if [[ -z "$projectName" ]]; then
...
-		create_compose_files
+		create_compose_files "$preferredRelease"
 		projects=$(get_existing_projects)
 		;;
 	esac
 
 	envCount=$(find . -name "*.env" -type f | wc -l)
 	if [ "$envCount" -gt 0 ]; then
 		while [[ -z "$projectName" ]]; do
+		  echo
```
Comment (garethbowen) [inline scripts/docker-helper-4.x/cht-docker-compose.sh:300]: Not needed?

Comment (garethbowen) [reply scripts/docker-helper-4.x/cht-docker-compose.sh:300]: ... or new line?

File: scripts/docker-helper-4.x/cht-docker-compose.sh (hunk 1 above)
Comment (mrjones-plip) [reply scripts/docker-helper-4.x/cht-docker-compose.sh]: ah - sorry about that!  fixed all tabs -> spaces in 2d5fff2c8

File: scripts/docker-helper-4.x/cht-docker-compose.sh (hunk 2 above)
Comment (mrjones-plip) [reply scripts/docker-helper-4.x/cht-docker-compose.sh]: yeah - I'm going for a new line here.  each of the questions felt really cramped, especially when we're going to show a multi-column output with all the branches/versions.  Open to another way of doing this but would prefer to keep the extra spaces!

Comment (garethbowen) [reply scripts/docker-helper-4.x/cht-docker-compose.sh]: Thanks... FYI I've just tried this script out to test a new SW library and it worked well!

**Review by njuguna-n — APPROVED** [review]
Good job on this.

**Comment by mrjones-plip** [comment]
Thanks @njuguna-n  , @garethbowen  and @tatilepizs !

> What do you think about asking about cht-core version when you start an instance that was already created? do you think that it might be confusing?

Oh - interesting! I think this feature is good as a quick win and your idea might be good for another round of development.  Let's deploy this so folks can use it and see what feedback we get back.  I'm hoping that being able to deploy two instances, one `master` and one a `branch` version, very quickly will obviate the need to change one instance at boot.  It will be easy enough to come back with another PR to add this at a later date if we deem it important!

**Comment by mrjones-plip** [comment]
Here's a demo of what the almost final UX is like:

https://github.com/medic/cht-core/assets/8253488/b5cde677-63c7-43f3-b599-47111dfa2064

**Pull request merged**